src/main/java/com/openlattice/postgres/mapstores/AbstractBasePostgresMapstore.java

```diff
@@ -182,8 +187,8 @@ public V load( K key ) {
             stmt.setFetchSize( 50000 );
             final ResultSet rs = stmt.executeQuery( selectAllKeysQuery );
             return StreamUtil
-                    .stream( () -> new KeyIterator<K>( rs,
-                            new CountdownConnectionCloser( connection, 1 ), this::mapToKey ) )
+                    .stream( Lists.newArrayList( new KeyIterator<K>( rs,
```

Comment:
This forces a read of all data at once instead of streaming from the database. What's the reasoning behind this change?

Reply:
yeah I wasn't really sure what the best thing to do here was -- things were broken because the connection was being closed before the ResultSet was read, and this was a fix, but if there's a better way to do it that keeps the connection alive then I'd be happy to change it.

Reply:
This forces a fetch of all the keys at once. Need to find a better way